File: algorithms/searching/binary-search/BinarySearch.c
int binarySearch(int arr[], int target) {
  int low = 0;
  int high = sizeof(arr) / sizeof(arr[0]);
  int mid;

  while (low <= high) {
    mid = low + (high - low) / 2;

    if (arr[mid] == target) {
      return mid;
    } else if (arr[mid] > target) {
      high = mid - 1;
    } else {
      low = mid + 1;
    }
  }

  return -1;
}

File: algorithms/searching/binary-search/BinarySearch.ts
class Searching {
  public static binarySearch(arr: number[], target: number): number {
    let low: number = 0;
    let high: number = arr.length - 1;
    let mid: number;

    while (high >= low) {
      mid = low + Math.floor((high - low) / 2);

      if (arr[mid] == target) return mid;
      else if (arr[mid] > target) high = mid - 1;
      else low = mid + 1;
    }

    return -1;
  }
}

export { Searching };
